src/gc.c

```diff
@@ -171,6 +159,16 @@ void GCContext_ForceBGInvoke(GCContext* gc) {
   GCContext_CommonForceInvoke(gc, NULL);
 }
 
+static void GCContext_UnblockClient(void* data) {
+  RedisModuleBlockedClient *bc = data;
+  RedisModule_BlockedClientMeasureTimeEnd(bc);
+  RedisModule_UnblockClient(bc, NULL);
+}
+
+void GCContext_WaitForAllOperations(RedisModuleBlockedClient* bc) {
+  redisearch_thpool_add_work(gcThreadpool_g, GCContext_UnblockClient, bc, THPOOL_PRIORITY_HIGH);
```

Comment:
Do we ever use any other priority then `THPOOL_PRIORITY_HIGH`? If so it might cause an issue no?

Reply:
Not with this thread pool, which has a single thread that prefers the high priority. Maybe we should add an API that doesn’t require us to specify the priority

Reply:
Probably a good idea but out of the scope of this PR I believe.